apollo-compiler/src/main/kotlin/com/apollographql/apollo3/compiler/codegen/java/model/ModelBuilder.kt

```diff
@@ -103,4 +112,42 @@ internal class ModelBuilder(
       }
     }
   }
+
+  private fun TypeSpec.addBuilder(context: JavaContext): TypeSpec {
+    val fields = fieldSpecs.filter { !it.modifiers.contains(Modifier.STATIC) }
+      .filterNot { it.name.startsWith(prefix = "$") }
+    if (fields.isEmpty()) {
+      return this
+    } else {
+      val builderVariable = JavaClassNames.Builder.simpleName().decapitalizeFirstLetter()
+      val builderClass = ClassName.get("", JavaClassNames.Builder.simpleName())
+      val toBuilderMethod = MethodSpec.methodBuilder(Builder.TO_BUILDER_METHOD_NAME)
+        .addModifiers(Modifier.PUBLIC)
+        .returns(builderClass)
+        .addStatement("\$T \$L = new \$T()", builderClass, builderVariable, builderClass)
+        .addCode(fields
+          .map { CodeBlock.of("\$L.\$L = \$L;\n", builderVariable, it.name, it.type.unwrapOptionalValue(it.name)) }
+          .fold(CodeBlock.builder()) { builder, code -> builder.add(code) }
+          .build()
+        )
+        .addStatement("return \$L", builderVariable)
+        .build()
+      val buildableTypes = typeSpecs.filter {
+        it.typeSpecs.find { it.name == JavaClassNames.Builder.simpleName() } != null
+      }.map { ClassName.get("", it.name) }
+
+      return toBuilder()
+        .addMethod(toBuilderMethod)
+        .addMethod(Builder.builderFactoryMethod())
+        .addType(
+          Builder(
+            targetObjectClassName = ClassName.get("", name),
+            fields = fields.map { it.name to it.type.unwrapOptionalType() },
```

Comment:
Also same as above: escape the property name:
```suggestion
            fields = fields.map { context.layout.propertyName(it.name) to it.type.unwrapOptionalType() },
```